lib/ClangImporter/ClangImporter.cpp

```diff
@@ -631,6 +631,17 @@ importer::getNormalInvocationArguments(
       });
     }
 
+    // To support -apple-none, -apple-none-macho, -unknown-none-wasm triples.
+    if (triple.getVendor() == llvm::Triple::VendorType::Apple) {
+      invocationArgStrs.insert(invocationArgStrs.end(), {"-D__APPLE__"});
+    }
+    if (triple.isOSBinFormatMachO()) {
+      invocationArgStrs.insert(invocationArgStrs.end(), {"-D__MACH__"});
+    }
+    if (triple.isOSBinFormatWasm()) {
+      invocationArgStrs.insert(invocationArgStrs.end(), {"-D__wasi__"});
```

Comment:
The macro here should be `__wasm__`.  WASI is not related to the object file format but the environment (WASI uses WASM and provides a system interface).

Reply:
Aha, so the reason why I seemingly needed to add `-D__wasi__` is that we have SwiftShims code that conditionalizes based on it, and we fail to build the embedded stdlib for `wasm-unknown-none-wasm` without this `-D__wasi__`. The concrete piece of code is here: https://github.com/apple/swift/blob/main/stdlib/public/SwiftShims/swift/shims/Visibility.h#L137

Sounds like you're saying we should fix Visibility.h instead?

Reply:
Yeah, that definitely looks like a bug.  That should be `__wasm__`.

Reply:
Fixed!